// swissarmy-input-react/test/mock.js
module.exports = {
    fields  : [
        {
	        id          : "One",
	        type        : "text",
	        className   : "anInput",
	        labelText   : "My input:",
	        placeholder : "Type here...",
	        value       : "I'm a hot mess!",
	        maxLength   : "23"
        },
        {
	        id          : "Two",
	        type        : "number",
	        className   : "anInput",
	        labelText   : "My input:",
	        placeholder : "Type here...",
	        value       : "0",
	        maxLength   : "23"
        },
        {
	        id          : "Three",
	        type        : "date",
	        className   : "anInput",
	        labelText   : "My input:",
	        placeholder : "Type here...",
	        value       : "",
	        maxLength   : "23"
        },
        {
	        id          : "Four",
	        type        : "select",
	        className   : "anInput",
	        labelText   : "My input:",
	        placeholder : "Type here...",
	        value       : [
                "Buffy",
                "the",
                "vampire",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayer",
                "slayeswedr"
            ],
            index       : 3,
	        maxLength   : "23"
        }
    ]
}
