// swissarmy-input-react/modules/inputInput.js
jQuery  = require("jquery");
var React               = require("react"),
    $                   = jQuery,
    datepicker          = require("jquery-ui/datepicker"),
    Chosen              = require("../bower_components/react-chosen/react-chosen"),
    taghubInputInput    = React.createClass(
        {
            render              : function() {
                switch(this.props.type) {
                    case "select":
                        var options = this.props.value.map(function(value) {
                                return (
                                    React.DOM.option(
                                        {
                                            value   : value
                                        },
                                        value
                                    )
                                )
                            });

                        return (
                            Chosen(
                                {
                                    id              : this.props.id,
                                    noResultsText   : "",
                                    value           : this.props.value[false],
                                    onChange        : function() {
                                        if (this.props.extraAction) {
                                            this.props.extraAction()
                                        }
                                    }.bind(this)
                                },
                                [
                                    options
                                ]
                            )
                        );
                        break;
                    case "multiselect":
                        break;
                    case "checkbox":
                        return (
                            React.DOM.input(
                                {
                                    id          : this.props.id,
                                    type        : "checkbox",
                                    className   : this.props.className,
                                    value       : this.props.value,
                                    onChange    : function() {
                                        this.props.handleValueChange(document.getElementById(this.props.id).checked);
                                        if (this.props.extraAction) {
                                            this.props.extraAction()
                                        }
                                    }.bind(this)
                                }
                            )
                        );
                        break;
                    default:
                        return (
                            React.DOM.input(
                                {
                                    id          : this.props.id,
                                    type        : (this.props.type === "date" || this.props.type === "select" || this.props.type === "multiselect") ?
                                        "text"
                                    :
                                        this.props.type,
                                    className   : this.props.className,
                                    placeholder : this.props.placeholder,
                                    value       : this.props.value,
                                    maxLength   : this.props.maxLength,
                                    onChange    : function() {
                                        this.props.handleValueChange(this.getDOMNode().value);
                                        if (this.props.extraAction) {
                                            this.props.extraAction()
                                        }
                                    }.bind(this)
                                }
                            )
                        );
                        break;
                }
            },
            componentDidMount   : function() {
                this.forceUpdate()
            },
            componentDidUpdate  : function() {
                switch(this.props.type) {
                    case "date":
                        $("#"+this.props.id).datepicker(
                            {
                                dateFormat  : "yy-mm-dd",
                                showAnim    : ""
                            }
                        );
                        $("#ui-datepicker-div").addClass("swissArmyInputCalendar").addClass("swissArmyInput");
                        break;
                    case "select":
                        $("#"+this.props.id+"_chosen").addClass("swissArmyInputSelect").addClass("swissArmyInput")
                        break;
                    case "multiselect":
                        break;
                    default:
                        break;
                }
            }
        }
    );

module.exports  = taghubInputInput;
