// swissarmy-input-react/swissArmyInput.js
var React               = require("react"),
    inputLabel          = require("./modules/inputLabel"),
    inputInput          = require("./modules/inputInput"),
    swissArmyInputStyle = require("./swissArmyInputStyle"),
    swissArmyInput      = React.createClass(
        {
            handleValueChange   : function(newValue) {
                this.setState(
                    {
                        value   : newValue
                    }
                )
            },
            getInitialState     : function() {
                return{
                    value   : this.props.value,
                    dummy   : 0
                }
            },
            addStyleString      : function(str) {
                var node    = document.createElement("style");

                node.innerHTML  = str;
                document.body.appendChild(node);
            },
            componentDidMount   : function() {
                this.addStyleString(swissArmyInputStyle.replace(/HAMBURGER/g,"\\2630"));
            },
            render              : function() {
                return (
                    React.DOM.li(
                        {
                            className   : (this.props.type === "checkbox") ?
                                    "swissArmyInput swissArmyInputCheck"
                                :
                                    "swissArmyInput"
                        },
                        [
                            (this.props.type !== "checkbox") ?
                                    inputLabel(
                                        {
                                            id          : this.props.id,
                                            labelText   : this.props.labelText
                                        }
                                    )
                                :
                                    "",
                            inputInput(
                                {
                                    id                  : this.props.id,
                                    type                : this.props.type,
                                    className           : this.props.className,
                                    placeholder         : this.props.placeholder,
                                    value               : this.state.value,
                                    maxLength           : this.props.maxLength,
                                    handleValueChange   : this.handleValueChange,
                                    extraAction         : this.props.extraAction || false
                                }
                            ),
                            (this.props.type === "checkbox") ?
                                    React.DOM.label(
                                        {
                                            htmlFor     : this.props.id
                                        },
                                        this.props.labelText
                                    )
                                :
                                    ""
                        ]
                    )
                )
            }
        }
    );

module.exports  = swissArmyInput;
